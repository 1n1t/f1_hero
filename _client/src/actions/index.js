import axios from 'axios';
import {FETCH_RACES, FETCH_USER, SELECT_RACE} from './types';

export const fetchUser = () => async dispatch => {
  const res = await axios.get('/api/current-user');
  dispatch({ type: FETCH_USER, payload: res });
};

export const setUser = user => dispatch => {
	dispatch({ type: FETCH_USER, payload: user });
};

export const fetchRaces = () => async dispatch => {
	const res = await axios.get('/api/races');
	dispatch({ type: FETCH_RACES, payload: res });
};

export const selectRace = (round) => (dispatch, getState, payload) => {
	dispatch({ type: SELECT_RACE, payload: round });
};
