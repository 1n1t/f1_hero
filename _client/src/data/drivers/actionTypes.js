/* eslint-disable import/prefer-default-export */
export const UPDATE_DRIVER_LIST = 'drivers/UPDATE_DRIVER_LIST';
