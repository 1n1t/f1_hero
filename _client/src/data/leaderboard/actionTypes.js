/* eslint-disable import/prefer-default-export */
export const UPDATE_SCORES = 'leaderboard/UPDATE_SCORES';
