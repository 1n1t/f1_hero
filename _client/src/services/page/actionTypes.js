/* eslint-disable import/prefer-default-export */
export const UPDATE_PAGE_HEADING = 'page/UPDATE_PAGE_HEADING';
