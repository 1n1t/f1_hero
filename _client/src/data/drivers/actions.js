/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

const updateDriverList = drivers => ({
	type: actionTypes.UPDATE_DRIVER_LIST,
	payload: drivers
});

export const getDriverList = () => async dispatch => {
	const response = await api.getDriverList();
	dispatch(updateDriverList(response.data.drivers));
};
