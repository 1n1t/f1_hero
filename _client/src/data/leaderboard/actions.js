/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

const getScores = scores => ({
	type: actionTypes.GET_SCORES,
	payload: scores
});

export const getUsersScores = () => async dispatch => {
	const response = await api.getUsersScores();
	dispatch(getScores(response.data.leaderboard));
};
