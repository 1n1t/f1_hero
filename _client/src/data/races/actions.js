/* eslint-disable import/prefer-default-export */
import * as api from './api';
import * as actionTypes from './actionTypes';

const updateRaceList = races => ({
	type: actionTypes.UPDATE_RACE_LIST,
	payload: races
});

export const selectRace = id => ({
	type: actionTypes.SELECT_RACE,
	payload: id
});

export const updateRace = ({ round, field, value }) => ({
	type: actionTypes.UPDATE_RACE,
	payload: { round, field, value }
});

export const getRaceList = () => async dispatch => {
	const response = await api.getRaceList();
	dispatch(updateRaceList(response.data.races));
};
