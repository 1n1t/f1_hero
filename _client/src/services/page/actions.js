/* eslint-disable import/prefer-default-export */
import * as actionTypes from './actionTypes';

export const updatePageHeading = heading => ({
	type: actionTypes.UPDATE_PAGE_HEADING,
	payload: heading
});
