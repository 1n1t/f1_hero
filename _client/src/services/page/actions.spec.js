import pageSample from 'JestClient/samples/page';
import { updatePageHeading } from './actions';
import { UPDATE_PAGE_HEADING } from './actionTypes';

it(`dispatches "${UPDATE_PAGE_HEADING}" action with correct payload`, () => {
	expect(updatePageHeading(pageSample.heading)).toMatchSnapshot();
});
