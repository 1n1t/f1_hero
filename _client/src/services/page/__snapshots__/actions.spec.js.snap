// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dispatches "page/UPDATE_PAGE_HEADING" action with correct payload 1`] = `
Object {
  "payload": "Foo Bar",
  "type": "page/UPDATE_PAGE_HEADING",
}
`;
